src/openzaak/components/catalogi/admin/zaaktypen.py

```diff
@@ -307,6 +307,23 @@ def get_related_objects(self, obj):
 
         return resource_list, id_list
 
+    def response_post_save_change(self, request, obj):
+        if "_publish" in request.POST and "_auto-publish" in request.POST:
+            # publish related types
+            for besluittype in obj.besluittypen.filter(concept=True):
+                besluittype.publish()
+
+            for iot in obj.informatieobjecttypen.filter(concept=True):
+                iot.publish()
+            self.message_user(
+                request,
+                _("Published related zaaktype objects"),
+                messages.INFO,
+                "autopublish",
+            )
```

Comment:
This does nothing and I do not know how to make it do something.

Reply:
It is cleared here: https://github.com/open-zaak/open-zaak/blob/e08dbbcf405c88322806764f60bf4643d52aea8f/src/openzaak/components/catalogi/admin/mixins.py#L62-L65

I'm not sure if I change it to allow the message without affecting other things.

Maybe add it after the super function is run.

Reply:
Seems like a hack, check with @stevenbal about this: https://github.com/open-zaak/open-zaak/commit/414ddafacff0f507de3d2f01baa96da7b9a49c36

Reply:
Better would be to also mention the names of the related entities which have been auto-published

Reply:
The clearing of the messages is a bit of a hack indeed, it was to prevent the default admin save messages from showing up. I couldn't find another way to suppress these default 

Adding the message after `super().response_post_save_change` has run would make it work, though if I remember correctly the actual publishing of the related resources must happen before the `ZaakType` is published.

Reply:
It was originally an aesthetic thing, but if I add the name of all the related published items, it might make more sense to display the published types bellow.

The question is: is it better to show the confirmation first then the auto-published items or the other way around.
For the former, I can just move the `self.message_user`.

Reply:
I just displayed the auto-published messages bellow the confirmation.